components/web/services/org.wso2.carbon.apimgt.authenticator/src/main/java/org/wso2/carbon/apimgt/authenticator/AuthenticatorAPI.java

```diff
@@ -146,9 +167,167 @@ public Response authenticate(@Context Request request, @FormDataParam ("username
+            Map<String, Scope> storeScopesMap = apiDefinitionFromSwagger20.getScopes(storeRestAPI);
+            final StringBuffer scopes = new StringBuffer();
+            storeScopesMap.keySet().forEach(key -> {
+                scopes.append(key).append(" ");
+            });
+            String storeScopes = scopes.toString();
+            storeScopes = storeScopes + KeyManagerConstants.OPEN_ID_CONNECT_SCOPE;
+            oAuthApplicationInfo = createDCRApplication(appContext.substring(1), grantTypes, validityPeriod);
+            if (oAuthApplicationInfo != null) {
+                String oAuthApplicationClientId = oAuthApplicationInfo.getClientId();
+                String oAuthApplicationCallBackURL = oAuthApplicationInfo.getCallBackURL();
+                String oAuthApplicationClientSecret = oAuthApplicationInfo.getClientSecret();
+
+                JsonObject oAuthData = new JsonObject();
+                oAuthData.addProperty("client_id" , oAuthApplicationClientId);
+                oAuthData.addProperty("callback_URL" , oAuthApplicationCallBackURL);
+                oAuthData.addProperty("scopes" , storeScopes);
+                oAuthData.addProperty("client_secret" , oAuthApplicationClientSecret);
+                APIMStoreConfigurations storeConfigs = new APIMStoreConfigurations();
+                oAuthData.addProperty("isSSOEnabled" , storeConfigs.getIsSSOEnabled());
+                return Response.status(Response.Status.OK).entity(oAuthData).build();
+            } else {
+                log.warn("No information available in OAuth application.");
+                return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                        .entity("Error while creating the OAuth application!").build();
+            }
+        } catch (APIManagementException e) {
+            ErrorDTO errorDTO = AuthUtil.getErrorDTO(e.getErrorHandler(), null);
+            log.error(e.getMessage(), e);
+            return Response.status(e.getErrorHandler().getHttpStatusCode()).entity(errorDTO).build();
+        }
+    }
+
+    /**
+     * This method requests the access token and redirects the user to a given URL.
+     *
+     */
+    @GET
+    @Path("/callback")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response callback(@Context Request request) {
+        String appContext = AuthUtil.getAppContext(request);
+        String appName = appContext.substring(1);
+        String restAPIContext;
+        APIMStoreConfigurations storeConfigs = new APIMStoreConfigurations();
+        if (appContext.contains("editor")) {
+            restAPIContext = AuthenticatorConstants.REST_CONTEXT + "/publisher";
+        } else {
+            restAPIContext = AuthenticatorConstants.REST_CONTEXT + appContext;
+        }
+        String grantType = "authorization_code";
+        String callbackURI = "https://localhost:9292/store/auth/apis/login/callback";
```

Comment:
This should come from a config, and it should be the same url which was used in /authorize call.